### ISSUE TITLE
Blending broken

### ISSUE THREAD

**Issue by davidmanzanares** [CartoDB/tangram-reference#7]
Mapnik expose many blending modes (also called *-comp-op).

Transparency is traditionally handled with the **over** operator, which I think is what Mapnik calls `src-over` and what Tangram calls `overlay`.

To make things worse, Mapnik also has an `overlay` operator (see https://tilemill-project.github.io/tilemill/docs/guides/comp-op/), and it seems that it is a much more complex operator. Although the previous documentation doesn't clearly state how exactly is that operation I suspect that it is not supported by WebGL because WebGL support for blending modes is very limited (see https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glBlendFunc.xml). Although there are some extensions that could implement it,  I guess that that is beyond current Tangram implementation.

I think we should expose the "src-over" (in Mapnik terms) capability, disable "overlay" (in Mapnik terms) from this reference. And check that we are doing the necessary translation from "src-over" to "overlay". 

Extra detail: tangram differences between `overlay` and `inlay`. The difference is the order in which features are drawn, and the order affects the final result, but the operation of overlay/inlay is the same as `src-over`.

To summarize:

| Mapnik name  | Tangram name | Supported|
| ------------- | ------------- | ------------- |
| src-over  | overlay/inlay  | No (works by default or by using "overlay" in CartoCSS, it is partially broken, but a pull request has been made) |
| multiply | multiply | Partially (Tangram implementation works only with opaque features) |
| plus | add | No (Tangram supports it partially, but the translation in tangram-carto is not present, a pull request in Tangram is needed to support additive blending with outlines)|
| * | Not supported in Tangram|

`src-over` can be fixed by translating the name.
`multiply` can only be fixed completely if we fix it in Tangram
`plus` can only be fixed completely if we fix it in Tangram.
`*` they could be fixed by adding functionality to Tangram.

**Comment by davidmanzanares** [comment]
While investigating more I've found:

-  Mapnik spec is based on SVG spec: https://www.w3.org/TR/2009/WD-SVGCompositing-20090430/
- Most Mapnik blending modes can be implemented in Tangram just by changing a few flags, and implementing the outline-fill blending in the shaders (this is needed because Tangram implementation is... surprising)
- The rest of the blending modes can be implemented in Tangram by using this WebGL extension https://www.khronos.org/registry/OpenGL/extensions/KHR/KHR_blend_equation_advanced.txt Using it is easy, and hardware/driver support is not perfect, but it's quite good: http://webcache.googleusercontent.com/search?q=cache:KM7lyHSR84cJ:feedback.wildfiregames.com/report/opengl/feature/GL_KHR_blend_equation_advanced+&cd=3&hl=en&ct=clnk&gl=es

**Comment by davidmanzanares** [comment]
I was wrong regarding the my last point. The extension exists, and it is quite supported, but it is an **OpenGL** extension, not a WebGL one. The set of WebGL extensions is very small (https://www.khronos.org/registry/webgl/extensions/) and it doesn't include this.

**Comment by IagoLast** [comment]
just for the record: with our current model the `blend-mode` value cannot depend on a filter.

It is the same case that `marker-allow-overlap` addressed in https://github.com/CartoDB/tangram-carto/issues/55

Should we document it and proceed like 'marker-allow-overlap' or is time to explore the `filters` approach [proposed by the tangram team](https://mapzen.com/documentation/tangram/Filters-Overview/)?

cc @rochoa @alonsogarciapablo

**Comment by davidmanzanares** [comment]
Current status is:

- Src-over works fine.
- Additive blending on markers with outline is broken, a PR has been sent.
- Multiply is disabled by tangram-reference. Tangram doesn't support it fully.
